### ISSUE TITLE
"state mismatch" on nextjs example

### ISSUE THREAD

**Issue by kylemac** [auth0/nextjs-auth0#26]
### Description

When attempting to follow the directions in the nextjs auth0 example readme.md ([link](https://github.com/zeit/next.js/tree/canary/examples/auth0)), i get a `state mismatch` error locally. i've verified that all the credentials in my `.env` are appropriate and up-to-date, and i still receive the error. 

it seems like there was a recent fix that centered around a similar error where that user saw it resolve their issues, however i'm still seeing `state mismatch` error persisting. ([reference to previous issue](https://github.com/zeit/next.js/issues/9003)) that @sandrinodimattia addressed. 

### Reproduction

- `npx create-next-app --example auth0 auth0` 
- `cd auth0`
- `npm install`
- `cp .env.template .env`
- fill in the details for the ENV variables with the data from the auth0 dashboard
- `npm run dev`
- `open localhost:3000`
- click on "login" link in top nav
- use google auth to login (same issue with github auth - haven't tried all others)
- land on `/api/callback?code=XXX&state=XXX` and see `state mismatch` error in view

### Environment

- **"@auth0/nextjs-auth0": "0.4.0",** per [example repo](https://github.com/zeit/next.js/tree/canary/examples/auth0) - same issue persisted with `0.5.0`. 
- **mac / chrome:**
- package.json...
```
{
  "name": "nextjs-auth0-example",
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start"
  },
  "author": "",
  "license": "MIT",
  "dependencies": {
    "@auth0/nextjs-auth0": "0.4.0",
    "dotenv": "^8.1.0",
    "isomorphic-unfetch": "^3.0.0",
    "next": "latest",
    "react": "^16.9.0",
    "react-dom": "^16.9.0"
  }
}
```

**Comment by sandrinodimattia** [comment]
I just tried all steps you described and everything worked as expected:

![image](https://user-images.githubusercontent.com/655409/67003192-bd8bb880-f0dd-11e9-8628-e9886a6f055b.png)

Could you verify the following:

1. When the user is redirect to `/api/login`, what state is being returned in the cookie?

![image](https://user-images.githubusercontent.com/655409/67003269-e0b66800-f0dd-11e9-9ef6-9dd9ae27a39e.png)

2. What state is persisted in the cookie?

![image](https://user-images.githubusercontent.com/655409/67003368-0a6f8f00-f0de-11e9-9b6b-e5db1c96cbad.png)

3. When the user is redirected from Auth0 to `/api/callback`, what state is being returned there?

![image](https://user-images.githubusercontent.com/655409/67003341-fa57af80-f0dd-11e9-9d7e-136518ceb8a0.png)

**Comment by kylemac** [comment]
for what it's worth, i noticed there were some small differences between your personal example (https://github.com/sandrinodimattia/nextjs-auth0-example.git) and the zeit/nextjs example (https://github.com/zeit/next.js/tree/master/examples/auth0) so i tried setting up your personal example and still got the `state mismatch` error.

**Comment by kgoggin** [comment]
FWIW I'm seeing the same issue @kylemac describes, with a pretty much identical setup. I first followed the instructions in the README to set up my existing nextjs app, and then tweaked it to match the example app - in both scenarios I get the state mismatch error.

**Comment by sandrinodimattia** [comment]
@kylemac @kgoggin I still haven't been able to reproduce this issue.

Would it be possible to show the following for `/api/login` and `/api/callback`:

 - The full URL
 - The request and response cookies
 - The persisted cookie under application state

![image](https://user-images.githubusercontent.com/655409/67610329-34861880-f792-11e9-84ef-b22346237757.png)

**Comment by sandrinodimattia** [comment]
Can you also test signing in here: https://nextjs-auth0-demo.now.sh/

Does that work for you or are you getting the same error?

**Comment by sandrinodimattia** [comment]
To speed things up you can also record [a HAR file](https://auth0.com/docs/troubleshoot/har) and send it to me at sandrino at auth0.com. Please make sure to reproduce this issue with non-production credentials since the HAR file will record every step of the sign in process.

**Comment by kylemac** [comment]
@sandrinodimattia signing in with this demo worked: https://nextjs-auth0-demo.now.sh/ 

Additionally, I had filed a support request and sent along a HAR file there. the ticket number is `#00440286`. I'd love if you could take a look because I believe the most recent response had an idea as to what is causing the issue and at first glance it was at the library-level, not at my implementation level.

**Comment by sandrinodimattia** [comment]
@kylemac thanks for the update. I reached out to my colleague who worked on the ticket and he mentioned that you are actually using a custom domain. When that is the case, you should use that in the `domain` setting.

So instead of:

```js
export default initAuth0({
  domain: 'acme.auth0.com',
  clientId: '<AUTH0_CLIENT_ID>'
```

You need to use the custom domain:

```js
export default initAuth0({
  domain: 'login.acme.com',
  clientId: '<AUTH0_CLIENT_ID>'
```

(This just serves as an example, replace it with your actual custom domain). 

Let me know if that helps.

**Comment by sandrinodimattia** [comment]
I also updated the guidance here: https://github.com/auth0/nextjs-auth0/commit/afaca845573a47c8980307e59570712abfc0f72d

**Comment by ian-speers** [comment]
Closing as done — please feel free to re-open this, or create a new issue if you need anything else. Thanks for your contribution!

**Comment by mkhoussid** [comment]
I'm not sure if this would help someone coming here in the future, but after hours of bashing my head, trying to figure out why I was able to login with Gmail but not Facebook, Github, etc, I entered `https://dev-tenant99.geo.auth0.com/login/callback` as my callback URL for Facebook and Github, and it started working. Might've been wrong on my part, but I understood the [docs](https://github.com/auth0/nextjs-auth0#auth0-configuration) to be saying, 'Enter `http://localhost:3000/api/callback` (or https://YOUR-app.com/api/callback) as the callback URL'.

**Comment by richrliu** [comment]
Hi, I am running into the same error when I set up Custom Domains. (No error when I turn off Custom Domains). The suggested guidance, which is including the `domain` setting, no longer applies since that is no longer a supported parameter of `initAuth0`. What is the new guidance for SDK version 3.5.0?

(I considered opening a new issue, but since it seemed to be related to Custom Domains and therefore isnt reproducible with the example app, I decided to comment instead. Let me know what is the right place for this issue).
